main/src/main/java/cgeo/geocaching/apps/navi/AbstractPointNavigationApp.java

```diff
@@ -36,24 +36,30 @@ protected AbstractPointNavigationApp(@NonNull final String name, @Nullable final
     }
 
     @Override
-    public void navigateWithoutSelector(@NonNull final Context context, @NonNull final Geocache cache) {
+    public void navigate(@NonNull final Context context, @NonNull final Geocache cache) {
         final Geopoint coords = cache.getCoords();
         assert coords != null; // asserted by caller
         navigate(context, coords);
     }
 
     @Override
-    public void navigate(@NonNull final Context context, @NonNull final Geocache cache) {
+    public void navigate(@NonNull final Context context, @NonNull final Waypoint waypoint) {
+        final Geopoint coords = waypoint.getCoords();
+        assert coords != null; // asserted by caller
+        navigate(context, coords);
+    }
+
+    @Override
+    public void navigateWithTargetSelector(@NonNull final Context context, @NonNull final Geocache cache) {
         final ArrayList<IWaypoint> targets = new ArrayList<>();
         targets.add(cache);
         for (final Waypoint waypoint : cache.getWaypoints()) {
-            final Geopoint coords = waypoint.getCoords();
```

Comment:
By changing the next two lines you also revert #14339 (which is not related to the reverted #14320)

Reply:
re-implemented